src/main/java/properties/ReadProperties.java

```diff
@@ -1,30 +1,44 @@
 
 package properties;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import java.io.*;
 import java.util.Properties;
 
-public class ReadProperties {
 
+public final class ReadProperties {
+
+    private static final Logger logger = LogManager.getLogger();
     private final Properties properties = new Properties();
+    private static ReadProperties instance;
+
+    public static ReadProperties getInstance() {
+        if (instance == null) {
+            instance = new ReadProperties();
+        }
+        return instance;
+    }
 
-    public ReadProperties() {
+    private ReadProperties() {
         try (InputStream inputStream = getClass().getClassLoader().getResourceAsStream("config.properties")) {
             properties.load(inputStream);
         } catch (IOException e) {
+            logger.debug("The file cannot be read...");
             e.printStackTrace();
         }
     }
 
     /***
      * Receives the key value from config.properties
-     * @param key
+     * @param key key from config.properties
      * @return key value
      */
-    public String getPropertyValueByKey(String key) {
-        if (properties.getProperty(key) != null) {
-            return properties.getProperty(key);
+    public String getKeyProperty(Type key) {
```

Comment:
In my opinion 'Type' is not the best name for property key because Type can relate to absolutely anything and it might confuse people

Reply:
renamed